cmd/limactl/hostagent.go

```diff
@@ -32,6 +33,10 @@ func newHostagentCommand() *cobra.Command {
 }
 
 func hostagentAction(cmd *cobra.Command, args []string) error {
+	//Without this the call to vz.StartGUI fails. Adding it here, as this has to be called before the vz cgo loads.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
```

Comment:
Can we do this only for vz mode?

Reply:
I tried options, but unfortunately could find a way to make it work. 

If i move this inside hostagent it will start to panic, because once within hostagent the vz cgo and all were loaded. So only if i add before cgo loads it works else ending up in panic.

I will try to add a limayaml check directly here and see if that works

Reply:
Done the changes. Now it should impact only vz driver with display="vz"
Sending if hostagent need to run a sync ui as a command arg. In start.go deciding if we need to run a gui based on driver. 

Do check it out and let me know if you have any feedback :)